src/api/transaction/submit.js

```diff
@@ -5,6 +5,16 @@ const validate = utils.common.validate;
 const Request = utils.common.core.Request;
 const convertErrors = utils.common.convertErrors;
 
+function convertSubmitErrors(callback) {
+  return function(error, data) {
+    if (data.engineResult !== 'tesSUCCESS') {
```

Comment:
This isn't right. See here: https://ripple.com/build/transactions/#result-categories

The immediate response only indicates an immediate failure if the result code is `tej` or `tem`. Anything else means the transaction _may_ appear in a ledger.